test/unit/index.js

```diff
@@ -155,47 +161,121 @@ describe('extractData()', function () {
-                    expect(data[key].pluralRuleFunction).to.be.a('function');
+                var data = extractData({
+                    locales    : ['en'],
+                    pluralRules: true,
                 });
+
+                expect(data.en).to.have.key('pluralRuleFunction');
+                expect(data.en.pluralRuleFunction).to.be.a('function');
             });
         });
 
         describe('relativeFields', function () {
             it('should contribute a `fields` object property', function () {
-                var data = extractData({relativeFields: true});
-                Object.keys(data).forEach(function (key) {
-                    expect(data[key]).to.have.key('fields');
-                    expect(data[key].fields).to.be.an('object');
+                var data = extractData({
+                    locales       : ['en'],
+                    relativeFields: true,
                 });
+
+                expect(data.en).to.have.key('fields');
+                expect(data.en.fields).to.be.an('object');
+            });
+        });
+    });
+
+    describe('Locale hierarchy', function () {
+        it('should determine the correct parent locale', function () {
+            var data = extractData({
+                locales: [
+                    'en',
+                    'pt-MZ',
+                    'zh-Hant-HK',
+                ]
+            });
+
+            expect(data['en']).to.not.have.key('parentLocale');
+
+            expect(data['pt-MZ']).to.have.key('parentLocale');
+            expect(data['pt-MZ'].parentLocale).to.equal('pt-PT');
+
+            expect(data['zh-Hant-HK']).to.have.key('parentLocale');
+            expect(data['zh-Hant-HK'].parentLocale).to.equal('zh-Hant');
+            expect(data['zh-Hant']).to.not.have.key('parentLocale');
+        });
+
+        it('should de-duplicate data with suitable ancestors', function () {
+            var locales = [
+                'es-AR',
+                'es-MX',
+                'es-VE',
+            ];
+
+            var data = extractData({
+                locales       : locales,
+                pluralRules   : true,
+                relativeFields: true,
+            });
+
+            expect(data['es-AR']).to.have.keys('parentLocale');
+            expect(data['es-AR']).to.not.have.keys('pluralRuleFunction', 'fields');
+
+            expect(data['es-MX']).to.have.keys('parentLocale', 'fields');
+            expect(data['es-MX']).to.not.have.keys('pluralRuleFunction');
+
+            expect(data['es-VE']).to.have.keys('parentLocale');
+            expect(data['es-VE']).to.not.have.keys('pluralRuleFunction', 'fields');
+
+            locales.forEach(function (locale) {
+                var pluralRuleFunction;
+                var fields;
+
+                // Traverse locale hierarchy for `locale` looking for the first
+                // occurrence of `pluralRuleFunction` and `fields`;
```

Comment:
Now that `parentLocale` has been added this tests makes sure it exists and works!